src/components/Heading/Heading.tsx

```diff
@@ -54,17 +52,12 @@ const Heading: React.FC<HeadingProps> & {
     node.style.setProperty("--heading-clamp-lines", ellipsisMaxLines.toString())
   );
   const finalStyle = useStyle(style, { color: customColor });
-  const classNames = cx(
-    styles.headingComponent,
-    className,
-    getStyle(styles, camelCase("element-type-" + type)),
-    getStyle(styles, camelCase("size-" + size)),
-    {
-      [styles.multiLineEllipsis]: ellipsis && ellipsisMaxLines > 1,
-      [styles.singleLineEllipsis]: ellipsis && ellipsisMaxLines <= 1,
-      [styles.suggestEditOnHover]: suggestEditOnHover
-    }
-  );
+  const typographyClassName = useLegacyHeadingClassNameByType(type, size);
+  const classNames = cx(styles.headingComponent, typographyClassName, className, {
+    [styles.multiLineEllipsis]: ellipsis && ellipsisMaxLines > 1,
+    [styles.singleLineEllipsis]: ellipsis && ellipsisMaxLines <= 1,
+    [styles.suggestEditOnHover]: suggestEditOnHover
+  });
```

Comment:
aren't these coming from the `Title` component?

Reply:
This is the legacy heading one, for avoiding weird bug and changes outside of our scope, I only changed the css and kept the funcunality  as it was before (our title component also support eliipsis and max lines... but I used here only classes)

Reply:
ok